kernel/system_work_q.c

```diff
@@ -26,10 +26,12 @@ static int k_sys_work_q_init(void)
 		.no_yield = IS_ENABLED(CONFIG_SYSTEM_WORKQUEUE_NO_YIELD),
 	};
 
+	k_sys_work_q.thread.base.user_options |= K_ESSENTIAL;
```

Comment:
This would not work as `k_work_queue_start()` calls `k_thread_create()` which eventually calls `z_init_thread_base()` which overrides `user_options`.